### PR TITLE
POC of client cert support

### PR DESCRIPTION
**Pull request by britcey**
re: #44  - Bearing in mind that I'm a Go novice and liberally borrowed a bunch of this code, here's a POC of client cert support, along with alternate CA support.

### CODE REVIEW

**Review by pyama86 — COMMENTED** [review]
Thanks your Contibute!
I have a  worry that It is mighn't work which don't use tls connection server?
If it could work when don't use tls connection and I will merge this PR!

**Comment by britcey** [comment]
With my (admittedly limited) testing, specifying TLSClientConfig when connecting to a plain http URL works just fine - the TLSClientConfig is just ignored.

I'll clean things up a bit more and update the PR.

File: netapp/netapp.go [hunk 1]
```diff
@@ -81,12 +85,16 @@ func NewClient(endpoint string, version string, options *ClientOptions) *Client
 		options.Timeout = 60 * time.Second
 	}
 
+	tlsConfig, err := NewTLSConfig(options)
+	if err != nil {
+		fmt.Println(err)
+		return nil
```
Comment (britcey) [inline netapp/netapp.go]: This should probably be the standard Go 'return nil, \<error\>' idiom vs. using Println, but that would involve changing the return signature and I didn't want this to be a breaking change.

**Comment by britcey** [comment]
I cleaned things up a bit and added some more tests;  other than the comment from above about the return value in NewClient, I think this is ready to go.

**Comment by britcey** [comment]
One more minor change - I bumped libraryVersion from '1' to '1.1' (could make that '1.1.0' in order to move to semantic versioning)

File: netapp/netapp.go (hunk 1 above)
Comment (pyama86) [reply netapp/netapp.go]: Thank you for your consideration. I will raise the version, so let's return an error without worrying about backward compatibility.

**Comment by britcey** [comment]
As discussed, updated NewClient to return any errors;  updated readme & tests as well.

I've bumped the version to '2.0.0' but of course you should set it as you see fit.

**Review by pyama86 — APPROVED** [review]
Thanks a lot!!!!

**Pull request merged**